### PR TITLE
Use +Inf label additionally for Histogram metrics

### PR DESCRIPTION
**Pull request by shtripat**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Motivation and Context
Issue: https://github.com/minio/minio/issues/18790

## How to test this PR?
`mc admin prometheus metrics ALIAS | grep ttfb` would show additional rows with label `+Inf` as below
```
# HELP minio_s3_requests_ttfb_seconds_distribution Distribution of time to first byte across API calls
# TYPE minio_s3_requests_ttfb_seconds_distribution gauge
minio_s3_requests_ttfb_seconds_distribution{api="getbucketlocation",le="+Inf",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketlocation",le="0.050",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketlocation",le="0.100",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketlocation",le="0.250",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketlocation",le="0.500",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketlocation",le="1.000",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketlocation",le="10.000",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketlocation",le="2.500",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketlocation",le="5.000",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketobjectlockconfiguration",le="+Inf",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketobjectlockconfiguration",le="0.050",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketobjectlockconfiguration",le="0.100",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketobjectlockconfiguration",le="0.250",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketobjectlockconfiguration",le="0.500",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketobjectlockconfiguration",le="1.000",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketobjectlockconfiguration",le="10.000",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketobjectlockconfiguration",le="2.500",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="getbucketobjectlockconfiguration",le="5.000",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="headbucket",le="+Inf",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="headbucket",le="0.050",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="headbucket",le="0.100",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="headbucket",le="0.250",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="headbucket",le="0.500",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="headbucket",le="1.000",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="headbucket",le="10.000",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="headbucket",le="2.500",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="headbucket",le="5.000",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="newmultipartupload",le="+Inf",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="newmultipartupload",le="0.050",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="newmultipartupload",le="0.100",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="newmultipartupload",le="0.250",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="newmultipartupload",le="0.500",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="newmultipartupload",le="1.000",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="newmultipartupload",le="10.000",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="newmultipartupload",le="2.500",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="newmultipartupload",le="5.000",server="127.0.0.1:9000"} 1
minio_s3_requests_ttfb_seconds_distribution{api="putobjectpart",le="+Inf",server="127.0.0.1:9000"} 306
minio_s3_requests_ttfb_seconds_distribution{api="putobjectpart",le="0.050",server="127.0.0.1:9000"} 306
minio_s3_requests_ttfb_seconds_distribution{api="putobjectpart",le="0.100",server="127.0.0.1:9000"} 306
minio_s3_requests_ttfb_seconds_distribution{api="putobjectpart",le="0.250",server="127.0.0.1:9000"} 306
minio_s3_requests_ttfb_seconds_distribution{api="putobjectpart",le="0.500",server="127.0.0.1:9000"} 306
minio_s3_requests_ttfb_seconds_distribution{api="putobjectpart",le="1.000",server="127.0.0.1:9000"} 306
minio_s3_requests_ttfb_seconds_distribution{api="putobjectpart",le="10.000",server="127.0.0.1:9000"} 306
minio_s3_requests_ttfb_seconds_distribution{api="putobjectpart",le="2.500",server="127.0.0.1:9000"} 306
minio_s3_requests_ttfb_seconds_distribution{api="putobjectpart",le="5.000",server="127.0.0.1:9000"} 306
```

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]

File: cmd/metrics-v2.go [hunk 1]
```diff
@@ -1714,6 +1715,17 @@ func getHistogramMetrics(hist *prometheus.HistogramVec, desc MetricDescription)
 				Value:          float64(b.GetCumulativeCount()),
 			}
 			metrics = append(metrics, metric)
+			// add metrics with +Inf label
```
Comment (krisis) [inline cmd/metrics-v2.go]: Shouldn't we add one `Inf` bucket per histogram (point in time) metric? It looks like we are adding one `Inf` bucket for every bucket here.

Comment (shtripat) [reply cmd/metrics-v2.go]: ok. I felt per API level might be needed. Will change to add at histogram level.

**Review by jiuker — APPROVED** [review]

File: cmd/metrics-v2.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/metrics-v2.go]: wait it is needed per API level @shtripat

Comment (harshavardhana) [reply cmd/metrics-v2.go]: what @krisis meant was that we must have +Inf per histogram - from the code change it looks like you are adding one Int per le value.

Comment (shtripat) [reply cmd/metrics-v2.go]: oh ok. something like below?
```
diff --git a/cmd/metrics-v2.go b/cmd/metrics-v2.go
index ca262f443..00e9f1448 100644
--- a/cmd/metrics-v2.go
+++ b/cmd/metrics-v2.go
@@ -1715,18 +1715,18 @@ func getHistogramMetrics(hist *prometheus.HistogramVec, desc MetricDescription)
                                Value:          float64(b.GetCumulativeCount()),
                        }
                        metrics = append(metrics, metric)
-                       // add metrics with +Inf label
-                       labels1 := make(map[string]string)
-                       for _, lp := range dtoMetric.GetLabel() {
-                               labels1[*lp.Name] = *lp.Value
-                       }
-                       labels1["le"] = fmt.Sprintf("%.3f", math.Inf(+1))
-                       metrics = append(metrics, Metric{
-                               Description:    desc,
-                               VariableLabels: labels1,
-                               Value:          float64(b.GetCumulativeCount()),
-                       })
                }
+               // add metrics with +Inf label
+               labels1 := make(map[string]string)
+               for _, lp := range dtoMetric.GetLabel() {
+                       labels1[*lp.Name] = *lp.Value
+               }
+               labels1["le"] = fmt.Sprintf("%.3f", math.Inf(+1))
+               metrics = append(metrics, Metric{
+                       Description:    desc,
+                       VariableLabels: labels1,
+                       Value:          dtoMetric.GetCounter().GetValue(),
+               })
```

Comment (harshavardhana) [reply cmd/metrics-v2.go]: @krisis ^^

**Review by krisis — APPROVED** [review]

**Pull request merged**